### PR TITLE
Improved documentation and TS definitions for colWidths and rowHeights options

### PR DESCRIPTION
**Pull request by swistach**
### Context
In #7507 @Maxoudela found problem with inconsistent TS definitions for `colWidths`.
Additionally, in JSDocs we didn't mention about limitation of using `autoColumnSiz`e along with `colWidths`.

This PR contains @Maxoudela suggestions. I added a test for TS definitions and extended the scope to changes for `rowHeights` option.

### How has this been tested?
1) Run `npm run test:types`

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #7507

### CODE REVIEW

File: src/dataMap/metaManager/metaSchema.js [hunk 1]
```diff
@@ -287,8 +287,14 @@ export default () => {
...
      *
+     * The default width for columns in the rendering process equals 50px.
+     *
+     * An `undefined` value is for detection in {@link Hooks#modifyColWidth} hook if plugin or setting changed the default size.
+     *
+     * Note: This option will forcely disable {@link AutoColumnSize} plugin.
+     *
      * @memberof Options#
-     * @type {number|number[]|string|string[]|Function}
+     * @type {number|number[]|string|string[]|undefined[]|Function}
```
Comment (Osky772) [inline src/dataMap/metaManager/metaSchema.js:297]: When I try to run `jsdoc` this file can't be parsed and throw an error in this line. After a change to `Array<undefined>` it's ok.

```suggestion
     * @type {number|number[]|string|string[]|Array<undefined>|Function}
```

File: src/dataMap/metaManager/metaSchema.js [hunk 2]
```diff
@@ -318,10 +324,13 @@ export default () => {
...
      * be set via either of those two plugins.
      *
+     * The default height for rows in the rendering process equals 23px.
      * Height should be equal or greater than 23px. Table is rendered incorrectly if height is less than 23px.
      *
+     * An `undefined` value is for detection in {@link Hooks#modifyRowHeight} hook if plugin or setting changed the default size.
+     *
      * @memberof Options#
-     * @type {number|number[]|string|string[]|Function}
+     * @type {number|number[]|string|string[]|undefined[]|Function}
```
Comment (Osky772) [inline src/dataMap/metaManager/metaSchema.js:333]: The same problem as above.

```suggestion
     * @type {number|number[]|string|string[]|Array<undefined>|Function}
```

**Review by Osky772 — CHANGES_REQUESTED** [review]
It looks good except a two comments

**Review by Osky772 — DISMISSED** [review]
All good 👌

**Comment by swistach** [comment]
Right before merging, I found there were unnecessary quote chars in the changelog file. Could you approve this PR one more time? 🙏 😄

**Review by Osky772 — APPROVED** [review]
All right 👍

**Pull request merged**